nibabel/brainvoyager/tests/test_bv.py

```diff
@@ -0,0 +1,315 @@
+                    ('a_string', b'test.txt'),
+                    ('nested_counter_integer', 2),
+                    ('fdr_table_info',
+                     [OrderedDict([('another_float', 0.0),
+                                   ('another_string', b'sample')]),
+                      OrderedDict([('another_float', 0.0),
+                                   ('another_string',
+                                    b'sample')])])])]),
+     ('another_nested_field', [])])
+
+TEST_HDR_PACKED = \
+    b''.join([b'\x01\xff\x06\x00\x01\x00\x00\x00\x03\x00\x00\x00\x04\x00\x00',
+              b'\x00\x00\x00\x00\x00\xd4\xc3\xb2\xa1\x00\x00\x80?HelloWorld!',
+              b'\x00\x00\x00\x00\x00\x01\x00\x00\x0033\xd3?test.txt\x00\x02',
+              b'\x00\x00\x00\x00\x00\x00\x00sample\x00\x00\x00\x00\x00sample',
+              b'\x00\x01\x00\x00\x0033\xd3?test.txt\x00\x02\x00\x00\x00\x00',
+              b'\x00\x00\x00sample\x00\x00\x00\x00\x00sample\x00\x01\x00\x00',
+              b'\x0033\xd3?test.txt\x00\x02\x00\x00\x00\x00\x00\x00\x00sample',
+              b'\x00\x00\x00\x00\x00sample\x00\x01\x00\x00\x0033\xd3?test.txt',
+              b'\x00\x02\x00\x00\x00\x00\x00\x00\x00sample\x00\x00\x00\x00',
+              b'\x00sample\x00'])
+
+
+def test_read_c_string():
+    # sample binary block
+    binary = b'test.fmr\x00test.prt\x00'
+    with InTemporaryDirectory():
+        # create a tempfile
+        path = 'test.header'
+        fwrite = open(path, 'wb')
+
+        # write the binary block to it
+        fwrite.write(binary)
+        fwrite.close()
+
+        # open it again
+        fread = open(path, 'rb')
+
+        # test readout of one string
+        assert_equal([s for s in read_c_string(fread)], [b'test.fmr'])
+
+        # test new file position
+        assert_equal(fread.tell(), 9)
+
+        # manually rewind
+        fread.seek(0)
+
+        # test readout of two strings
+        assert_equal([s for s in read_c_string(fread, 2, rewind=True)],
+                     [b'test.fmr', b'test.prt'])
+
+        # test automatic rewind
+        assert_equal(fread.tell(), 0)
+
+        # test readout of two strings with trailing zeros
+        assert_equal([s for s in read_c_string(fread, 2, strip=False)],
+                     [b'test.fmr\x00', b'test.prt\x00'])
+
+        # test new file position
+        assert_equal(fread.tell(), 18)
+
+        # test readout of one string from given position
+        fread.seek(0)
+        assert_equal([s for s in read_c_string(fread, start_pos=9)],
+                     [b'test.prt'])
+        fread.close()
+
+
+def test_combine_st():
+    vmr = BvVmrImage.from_filename(vmr_file)
+    STarray = []
+    for st in range(vmr.header._hdr_dict['nr_of_past_spatial_trans']):
+        STarray.append(parse_st(
+            vmr.header._hdr_dict['past_spatial_trans'][st]))
+    STarray = np.array(STarray)
+    combinedST = combine_st(STarray, inv=True)
+    correctCombinedST = [[1., 0., 0., 0.],
```

Comment:
Good catch: it does not. However, as I do not have more test files right now and no access to a BrainVoyager license at the moment, I need to search for other means to solve this. I will think about it.